include/lbann/utils/dnn_lib/miopen/dropout.hpp

```diff
@@ -43,13 +44,15 @@ using namespace miopen;
 
 inline size_t get_dropout_states_size()
 {
+  BASIC_PROF_REGION("miopen:get_dropout_states_size");
   size_t size;
   CHECK_MIOPEN(miopenDropoutGetStatesSize(get_handle(), &size));
```

Comment:
@benson31 I am wondering if this type of call on the dropout also has a similar cost to the get convolution workspace size call.

Reply:
I think it's worth investigating deeper, but not in this PR. The main focus here was adding the annotations. Adding these annotations highlighted the severe cost of `miopenConvolution*GetWorkSpaceSize`. Comparing to the cuDNN code path suggested a simple and acceptable fix. 

Regarding this specific function, the cuDNN code path will just call `cudnnDropoutGetStatesSize`. This matches that, and I don't see these as a hotspot (or even mildly concerning) in the context of other performance issues.